contrib/internet-urllib3.py

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python
+
+import urllib3
+import certifi
+
+def isInternetConnected(url="www.ietf.org"):
+  result = False
+  http = urllib3.PoolManager(
+    cert_reqs='CERT_REQUIRED', # Force certificate check.
+    ca_certs=certifi.where(),  # Path to the Certifi bundle.
+  )
+  try:
+    r = http.request('HEAD', 'https://' + url)
+    result = True
+  except Exception as e:  # urllib3.exceptions.SSLError
+    result = False
```

Comment:
So, if anything gets wrong while validating the CA CERT, we would wrongly assume that the connection is lost.

The more I think about a strategy to check internet connectivity the more I'm convinced there's no correct way to do this.